### PR TITLE
refactor: Add `crm` type to app credentials 

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This is a part of our bigger CRM refactor.

Currently, our CRM apps have the type of `_other_calendar` which gets processed with our other calendar apps. This PR aims to introduce the `_crm` type to credentials in preparation for a CRM manager. Since this is a bigger change I wanted it to be it's own PR to make reviewing easier.

### Changes
- Moved Zoho Bigin under the CRM category
- Removed Sendgrid from the CRM category (still being treated as `other_calendar`
- Changed CRM apps metadata to
    - Variant to "crm"
    - Type to "{appName}_crm"
- For querying installed apps with metadata we are now matching apps based on their slug and not their type
- CRM apps finally are listed under the CRM category (this one bugged me a lot!)

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is a UI/UX design document, please, share it here.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- Install each CRM and ensure they're being read on new bookings

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by joeauyeung** [comment]
* **#14147** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/14147?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @joeauyeung and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/web/pages/apps/installed/[category].tsx [hunk 1]
```diff
@@ -69,11 +69,19 @@ const IntegrationsContainer = ({
...
       success={({ data }) => {
         if (!data.items.length) {
+          const emptyHeaderCategory = (() => {
+            if (variant) {
+              return variant === "crm" ? t("crm") : t(variant).toLowerCase();
+            } else {
+              return t("other").toLowerCase();
+            }
+          })();
+
```
Comment (joeauyeung) [inline apps/web/pages/apps/installed/[category].tsx:79]: This fixes the empty screen from "No crm apps" to "No CRM apps"

**Review by sean-brydon — COMMENTED** [review]
CleanShot 2024-03-20 at 10 22 11
CleanShot 2024-03-20 at 10 22 15 - This is the badge on the install page

Tested everything works as i expected. Little NIT should we also transform these to be uppercase like the empty screen?

**Comment by joeauyeung** [comment]
> CleanShot 2024-03-20 at 10 22 11 CleanShot 2024-03-20 at 10 22 15 - This is the badge on the install page
> 
> Tested everything works as i expected. Little NIT should we also transform these to be uppercase like the empty screen?

@sean-brydon is this on "/apps/installed"? I'm seeing CRM being capitalized.
image

**Comment by sean-brydon** [comment]
> > CleanShot 2024-03-20 at 10 22 11 CleanShot 2024-03-20 at 10 22 15 - This is the badge on the install page
> > Tested everything works as i expected. Little NIT should we also transform these to be uppercase like the empty screen?
> 
> @sean-brydon is this on "/apps/installed"? I'm seeing CRM being capitalized. image

It was yeah - i am no longer seeing this wtf 😆

**Review by sean-brydon — DISMISSED** [review]
LGTM

**Comment by sean-brydon** [comment]
LGTM again - just re-running CI as e2e failing. Doesnt seem relevant to this PR , they may be flaky

**Review by sean-brydon — APPROVED** [review]

**Comment by zomars** [comment]
@sean-brydon @joeauyeung Code LGTM. But how are we going to update the App records in the DB without a proper migration?

**Pull request merged**